ext/curb_easy.c

```diff
@@ -973,21 +973,21 @@ static VALUE ruby_curl_easy_proxy_type_get(VALUE self) {
 
 #if defined(HAVE_CURLAUTH_DIGEST_IE)
 #define CURL_HTTPAUTH_STR_TO_NUM(node) \
-  (!strncmp("basic",node,5)) ? CURLAUTH_BASIC : \
-  (!strncmp("digest_ie",node,9)) ? CURLAUTH_DIGEST_IE : \
-  (!strncmp("digest",node,6)) ? CURLAUTH_DIGEST : \
-  (!strncmp("gssnegotiate",node,12)) ? CURLAUTH_GSSNEGOTIATE : \
-  (!strncmp("ntlm",node,4)) ? CURLAUTH_NTLM : \
-  (!strncmp("any",node,3)) ? CURLAUTH_ANY : \
-  (!strncmp("anysafe",node,7)) ? CURLAUTH_ANYSAFE : 0
+  (!strcmp("basic",node)) ? CURLAUTH_BASIC : \
+  (!strcmp("digest_ie",node)) ? CURLAUTH_DIGEST_IE : \
+  (!strcmp("digest",node)) ? CURLAUTH_DIGEST : \
+  (!strcmp("gssnegotiate",node)) ? CURLAUTH_GSSNEGOTIATE : \
+  (!strcmp("ntlm",node)) ? CURLAUTH_NTLM : \
+  (!strcmp("any",node)) ? CURLAUTH_ANY : \
+  (!strcmp("anysafe",node)) ? CURLAUTH_ANYSAFE : 0
 #else 
 #define CURL_HTTPAUTH_STR_TO_NUM(node) \
-  (!strncmp("basic",node,5)) ? CURLAUTH_BASIC : \
-  (!strncmp("digest",node,6)) ? CURLAUTH_DIGEST : \
-  (!strncmp("gssnegotiate",node,12)) ? CURLAUTH_GSSNEGOTIATE : \
-  (!strncmp("ntlm",node,4)) ? CURLAUTH_NTLM : \
-  (!strncmp("any",node,3)) ? CURLAUTH_ANY : \
-  (!strncmp("anysafe",node,7)) ? CURLAUTH_ANYSAFE : 0
+  (!strcmp("basic",node)) ? CURLAUTH_BASIC : \
```

Comment:
`strncmp("any", "anysafe", 3)` returns 0, so `CURLAUTH_ANY` was used for `:anysafe`.

I have updated the pull request with a less intrusive bugfix: check for "anysafe" first, then for "any".